tests/backend-storage/src/test/ServerStorage.test.ts

```diff
@@ -304,6 +304,49 @@ describe(`${ServerStorage.name}: ${serverStorage.constructor.name}`, () => {
     });
   });
 
+  describe(`${serverStorage.listDirectories.name}()`, () => {
+    it("should list directories", async () => {
+      const testDirectory1: TestRemoteDirectory =
+        await testDirectoryManager.createNew();
+      const testDirectory2: TestRemoteDirectory =
+        await testDirectoryManager.createNew();
+      const testDirectory3: TestRemoteDirectory =
+        await testDirectoryManager.createNew();
+      await testDirectory1.uploadFile(
+        { objectName: "reference1" },
+        undefined,
+        undefined
+      );
+      await testDirectory2.uploadFile(
+        { objectName: "reference2" },
+        undefined,
+        undefined
+      );
+      await testDirectory3.uploadFile(
+        { objectName: "reference3" },
+        undefined,
+        undefined
+      );
+      const queriedReferences = await serverStorage.listDirectories();
+
+      const queriedReference1 = queriedReferences.find(
```

Comment:
Would be good to have an assertion for the length of `listDirectories`. Without it it could return extra unexpected directories in the result.

Reply:
It initally had this check but I had to remove it since in the oss cloud there were straggling directories. Had a discussion about this with @Animagne.